### PR TITLE
Fix/post filter unit tests

### PR DESCRIPTION
**Pull request by saminahbab**
### What

Fixed broken unit test for graceful service shutdown 
Added Kafka Producer Mock with Close function 

### How to review

run `make test` and `make test-component`

### Who can review

Team B

### CODE REVIEW

File: service/service_test.go [hunk 1]
```diff
@@ -227,33 +232,33 @@ func TestClose(t *testing.T) {
...
+							Server:      timeoutServerMock,
+							HealthCheck: hcMock,
+						}
+
+						err = svc.Close(context.Background())
+						So(err, ShouldNotBeNil)
+						So(err.Error(), ShouldResemble, "context deadline exceeded")
+						So(len(hcMock.StopCalls()), ShouldEqual, 1)
+						So(len(timeoutServerMock.ShutdownCalls()), ShouldEqual, 1)
+					})*/
```
Comment (PauloPortugal) [inline service/service_test.go:262]: Please remove the added tab/space.
Do we need it?

**Review by PauloPortugal — COMMENTED** [review]
It would be nice to include an explanation on the commit body, on why this failed and what was the fix for it.

This will help other developers understand a similar issue in other services.

File: service/service_test.go (hunk 1 above)
Comment (saminahbab) [reply service/service_test.go:262]: Gofmt does get a bit confused on my machine tabbing commented code.
I would be happy to remove the commented block, as that is cleaner. Would you prefer that?

Comment (PauloPortugal) [reply service/service_test.go:262]: Yes I rather remove commented code.

Comment (saminahbab) [reply service/service_test.go]: Commented out code removed.

**Comment by saminahbab** [comment]
@PauloPortugal https://github.com/ONSdigital/dp-cantabular-filter-flex-api/pull/43/commits/4de95d226641b0e727836e9cf45fa5ef54f87790 now has a commit message describing the reason for the fix.

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**